addons/l10n_ch/models/mail_template.py

```diff
@@ -31,7 +31,9 @@ def generate_email(self, res_ids, fields):
                 new_attachments = []
 
                 include_qr_report = 'l10n_ch.l10n_ch_qr_report' not in self.env.context.get('l10n_ch_mail_skip_report', [])
-                if include_qr_report and record.move_type == 'out_invoice' and record.partner_bank_id._eligible_for_qr_code('ch_qr', record.partner_id, record.currency_id):
+                if include_qr_report and record.move_type == 'out_invoice' and \
+                        record.partner_bank_id._eligible_for_qr_code('ch_qr', record.partner_id, record.currency_id) and \
+                        record.display_qr_code:
```

Comment:
```suggestion
                if include_qr_report and \
                        record.move_type == 'out_invoice' and \
                        record.partner_bank_id._eligible_for_qr_code('ch_qr', record.partner_id, record.currency_id) and \
                        record.display_qr_code:
```
Small detail, but I would put one test per line then (just to be consistent) :wink: